gistUpload.js

```
@@ -13,23 +13,30 @@ const HANG_PERCENT = 0.25;
  * @param {string} content - Contents of file from readFile
  * @param {string} gist - Name of gist supplied by user
  */
-function uploadFile(content, gist) {
-  return axios({
-    method: 'post',
-    url: API_URL,
-    auth: {
-      user: process.env.USER,
-      password: process.env.SECRET_TOKEN,
-    },
-    data: { files: { [gist]: { content } } },
-  })
-    .then((response) => {
-      console.log(`\nYour gist is done uploading.\nView it at: ${response.data.html_url}`);
-      return (response.data.html_url);
-    })
-    .catch(error => (console.log(error.response.data)));
+function uploadFile(contents, gist) {
+  try {
+    if (!gist || !contents) {
+      throw new Error('You must provide a file and a name.');
+    } else if (typeof contents !== 'string') {
+      throw new Error('Your file must contain a string.');
+    } else if (typeof gist !== 'string') {
+      throw new Error('Your gist name must be a string.');
+    } else {
+      const post = { [gist]: { content: contents } };
+      return axios.post(
+        API_URL,
+        { files: post },
+      )
+        .then((response) => {
+          console.log(`\nYour gist is done uploading.\nView it at: ${response.data.html_url}`);
+          return (response.data.html_url);
+        })
+        .catch(error => error);
```

Don't discard errors

---

done